### DISCUSSION TITLE
파싱 순서에 대한 논의

### DISCUSSION THREAD

**Discussion by JuneParkCode** [png-minishell/pngshell#7] [General]
1. 커맨드를 유저로부터 받는다.
2. 해당 커맨드를 환경 변수를 통해 치환한다
2-1. 이 때, envp[0] == NULL 일 때는 set을 함수의 envp 매개변수로 넣는다.
3. 치환된 커맨드를 parser에 던진다.

**Comment by 42mingylee** [comment]
envp[0] == NULL 일 때 set에서 받지만 
명령어 하나라도 export 됐을 경우 어떻게 돌아가는지 한번 확인 해봐야 할 듯...
env의 개수가 하나라도 증가하면 다르게 적용 될 것 같음.... 😢
set을 들고 있다가 find_key 에서 없을 때 set에 있는 PATH를 넣고 unset 할 때도 env와 set 모두 제거 해야됨...

**Reply by JuneParkCode** [reply]
맞는듯...
`envp[0] == NULL` 일 때도 쉘 시작과 함께 `env` 에 `pwd`, `shell level`이 추가됨
`env_substituter` 함수 자체를 `set`, `envp` 모두 조회하게끔 변경할 필요가 있음.

**Reply by JuneParkCode** [reply]
해당 문제를 처리할 수 있는 함수로 수정함!
